packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -64,6 +64,20 @@ function run (ipc, configFile, projectRoot) {
     return true
   }
 
+  const isValidComponentConfig = (config) => {
```

Comment:
Would it be more appropriate to update the existing ` isTestingTypeConfigured ` check in the ProjectLifeCycleManager to ensure the component test can't be loaded if this configuration is invalid?

https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/data/ProjectLifecycleManager.ts#L1251

Reply:
The issue there is that we can't reliably check for `Promise<{devServer}>` case from the parent process, since the config object gets serialized & sent back before that resolves, and that wouldn't get transferred over the IPC